sound/soc/sof/sof-audio.c

```diff
@@ -75,6 +75,10 @@ int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 			err = ret;
 	}
 
+	/* clear pipeline complete */
+	if (swidget->id == snd_soc_dapm_scheduler)
+		swidget->spipe->complete = 0;
```

Comment:
Can a pipeline belong to different paths?

Reply:
@ujfalusi yes it can and the scheduler widget will be freed only when the last of all the users lets it go